report/report.go

```diff
@@ -360,6 +369,61 @@ func (r Report) Upgrade() Report {
 	return cp
 }
 
+// RequiresUpgrade returns whether the report needs upgrading
+func (r Report) RequiresUpgrade(appVersion string) bool {
```

Comment:
Feeding the app version through here, and the whole `requiresUpgradeRevisionCheck()` logic, is overkill. If we get a report from a probe with a git revision then we should always upgrade. That will only affect us, for probes we are running in dev and during local development.